bundles/org.eclipse.cdt.lsp.editor.ui/src/org/eclipse/cdt/lsp/editor/ui/preference/LspEditorPreferencePage.java

```diff
@@ -37,7 +48,52 @@ public void init(IWorkbench workbench) {
 	@Override
 	public void createFieldEditors() {
 		PreferenceMetadata<Boolean> prefer = LspEditorPreferences.getPreferenceMetadata();
-		addField(new BooleanFieldEditor(prefer.identifer(), prefer.name(), getFieldEditorParent()));
+		var booleanFieldEditor = new BooleanFieldEditor(prefer.identifer(), prefer.name(), getFieldEditorParent());
+		addField(booleanFieldEditor);
+		
+		serverPath = new FileFieldEditor(LspEditorPreferences.SERVER_PATH,
+				LspEditorUiMessages.LspEditorPreferencePage_server_path, getFieldEditorParent());
+		addField(serverPath);
+
+		
+		serverOptions = new MultiLineTextFieldEditor(LspEditorPreferences.SERVER_OPTIONS,
+				LspEditorUiMessages.LspEditorPreferencePage_server_options, getFieldEditorParent());
+		addField(serverOptions);
+	}
+		
+	@Override
+	public boolean performOk() {
+		writeServerPathToProvider(serverPath.getStringValue());
+		writeServerOptionsToProvider(serverOptions.getStringValue());
+		return super.performOk();
+	}
+	
+	private void writeServerPathToProvider(String path) {
+		if (path == null || path.isBlank()) {
+			return;
+		}
+		List<String> commands = LspPlugin.getDefault().getCLanguageServerProvider().getCommands();
+		if (commands != null && !commands.isEmpty()) {
+			commands.set(0, path);
+		} else if (commands == null) {
+			commands = new ArrayList<>();
+			commands.add(path);
+		}
+		LspPlugin.getDefault().getCLanguageServerProvider().setCommands(commands);
+	}
+	
+	private void writeServerOptionsToProvider(String options) {
+		if (options == null || options.isBlank()) {
+			return;
+		}
+		List<String> commands = LspPlugin.getDefault().getCLanguageServerProvider().getCommands();
+		if (commands != null && !commands.isEmpty()) {
+			String serverPath = commands.get(0); // save server path
+			commands.clear(); // clear all old options
+			commands.add(serverPath);
+			commands.addAll(Arrays.asList(CommandLineUtil.argumentsToArray(options)));
+		}
+		LspPlugin.getDefault().getCLanguageServerProvider().setCommands(commands);
```

Comment:
Perhaps there is a future plan for the extra complication about how new settings are processed, but I think this is sufficient?

```suggestion
	@Override
	public boolean performOk() {
		writeSettingsToProvider(serverPath.getStringValue(), serverOptions.getStringValue());
		return super.performOk();
	}
	
	private void writeSettingsToProvider(String path, String options) {
		if (path == null || path.isBlank() || options == null) {
			// The validation of the field editors should have prevented invalid entries
			return;
		}
		List<String> commands = new ArrayList<>();
		commands.add(path);
		commands.addAll(Arrays.asList(CommandLineUtil.argumentsToArray(options)));
		LspPlugin.getDefault().getCLanguageServerProvider().setCommands(commands);
```

Reply:
This was meant as a start point for editing server settings. The settings could be processed with a focus on a change detection and to add/remove only the changes in the future.